go/ct/rlz/domains.go

```diff
@@ -385,3 +385,33 @@ func (gasDomain) SamplesForAll(as []vm.Gas) []vm.Gas {
 
 	return res
 }
+
+////////////////////////////////////////////////////////////
+// BlockNumber Range Domain
+
+type BlockNumberOffsetDomain struct{}
+
+func (BlockNumberOffsetDomain) Equal(a int64, b int64) bool { return a == b }
+func (BlockNumberOffsetDomain) Less(a int64, b int64) bool  { return a < b }
+func (BlockNumberOffsetDomain) Predecessor(a int64) int64   { return a - 1 }
+func (BlockNumberOffsetDomain) Successor(a int64) int64     { return a + 1 }
+func (BlockNumberOffsetDomain) SomethingNotEqual(a int64) int64 {
+	return a + 1
+}
+
+func (d BlockNumberOffsetDomain) Samples(a int64) []int64 {
+	return d.SamplesForAll([]int64{a})
+}
+
+func (BlockNumberOffsetDomain) SamplesForAll(as []int64) []int64 {
+	res := []int64{math.MinInt64, -1, 0, 1, 255, 256, 257, math.MaxInt64}
+
+	// Test every element off by one.
+	for _, a := range as {
+		res = append(res, a-1)
+		res = append(res, a)
+		res = append(res, a+1)
+	}
```

Comment:
This is a little optimization, we could remove duplicates. Or is this done somewhere else?